### PR TITLE
Added Documentation for Functions

### PR DESCRIPTION
**Pull request by Alonski**

### CODE REVIEW

File: contracts/MultiSend.sol [hunk 1]
```diff
@@ -1,74 +1,117 @@
+// Copyright (C) 2018 Alon Bukai This program is free software: you 
+// can redistribute it and/or modify it under the terms of the GNU General 
+// Public License as published by the Free Software Foundation, version. 
+// This program is distributed in the hope that it will be useful, 
+// but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+// or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+// more details. You should have received a copy of the GNU General Public
+// License along with this program. If not, see http://www.gnu.org/licenses/
```
Comment (Quazia) [inline contracts/MultiSend.sol:8]: I loves me some GPL.

File: contracts/MultiSend.sol [hunk 2]
```diff
@@ -1,74 +1,117 @@
...
 pragma solidity ^0.4.18;
 
 import "giveth-common-contracts/contracts/Escapable.sol";
 
-// TightlyPacked is cheaper if you need to store input data and if amount is less than 12 bytes.
-// Normal is cheaper if you don't need to store input data or if amounts are greater than 12 bytes.
-// Supports deterministic deployment. As explained here: https://github.com/ethereum/EIPs/issues/777#issuecomment-356103528
+/// @notice MultiSend is a contract for sending multiple ETH/ERC20 Tokens to multiple addresses
+/// In addition this contract can call multiple contracts with multiple amounts
+/// There are also TightlyPacked functions which in some situations allow for gas savings
```
Comment (Quazia) [inline contracts/MultiSend.sol]: I would include the parts that specify when each approach is cheaper, that's important information. Also, I would use ` to make `MultiSend` look all code-y

File: contracts/MultiSend.sol [hunk 3]
```diff
@@ -1,74 +1,117 @@
...
+  /// @notice Constructor using Escapable and Hardcoded values
   function MultiSend() Escapable(CALLER, DESTINATION) public {}
   
-  function multiTransferTightlyPacked(bytes32[] _addressAndAmount) payable public returns(bool) {
-    for (uint i = 0; i < _addressAndAmount.length; i++) {
-        _safeTransfer(address(_addressAndAmount[i] >> 96), uint(uint96(_addressAndAmount[i])));
+  /// @notice Send to multiple addresses using a byte32 array which
+  /// includes the address and the amount.
+  /// Payable
+  /// @param _addressesAndAmounts Bitwise packed array of addresses and amounts
```
Comment (Quazia) [inline contracts/MultiSend.sol]: Try and include some information on where the address is and where the amounts are. The address is stored in the 20 most significant bytes of the byte32 object and the amount is stored in the 12 least significant bytes of the byte32 object.

File: contracts/MultiSend.sol [hunk 4]
```diff
@@ -1,74 +1,117 @@
...
       }
       return true;
   }
 
-  function multiERC20TransferTightlyPacked(ERC20 _token, bytes32[] _addressAndAmount) public returns(bool) {
-      for (uint i = 0; i < _addressAndAmount.length; i++) {
-          _safeERC20Transfer(_token, address(_addressAndAmount[i] >> 96), uint(uint96(_addressAndAmount[i])));
+  /// @notice Send ERC20 tokens to multiple contracts 
+  /// using a byte32 array which includes the address and the amount.
+  /// @param _addressesAndAmounts Bitwise packed array of addresses and token amounts
```
Comment (Quazia) [inline contracts/MultiSend.sol]: missing a param here

File: contracts/MultiSend.sol [hunk 5]
```diff
@@ -1,74 +1,117 @@
...
+  /// @param _amounts Array of token amounts to send
+  function multiERC20Transfer(ERC20 _token, address[] _addresses, uint[] _amounts) public returns(bool) {
+      for (uint i = 0; i < _addresses.length; i++) {
+          _safeERC20Transfer(_token, _addresses[i], _amounts[i]);
       }
       return true;
   }
 
-  function _safeTransfer(address _to, uint _amount) internal {
+  function _safeTransfer(address _to, uint _amounts) internal {
```
Comment (Quazia) [inline contracts/MultiSend.sol]: Why is this plural here?

**Review by Quazia — CHANGES_REQUESTED** [review]
Just a few small things, I tried to only mention each issue once but that doesn't mean it's the only place it's present.

File: contracts/MultiSend.sol (hunk 1 above)
Comment (Alonski) [reply contracts/MultiSend.sol:8]: I `stole` this from you 😄

**Comment by Alonski** [comment]
@Quazia Is this ready to be merged?

File: contracts/MultiSend.sol (hunk 1 above)
Comment (Quazia) [reply contracts/MultiSend.sol:8]: It's not stealing if it's GPL licensed 😉

**Pull request merged**